### DISCUSSION TITLE
ActionSheetState and AlertState for UIKit

### DISCUSSION THREAD

**Discussion by larryonoff** [pointfreeco/swift-composable-architecture#561] [Ideas]
Composable Architecture needs more UIKit support. So I suggest extending `ActionSheetState` and `AlertState` to support `UIAlertController`, i.e.

1. Remove `ActionSheetState` operation systems limitations.
2. Move `ActionSheetState` and `AlertState` into Common folder.
3. Add convenient initialisers to `UIAlertController`, e.g. 

```swift
extension UIAlertController {

public static func actionSheet<Action>(
  state: ActionSheetState<Action>,
  send: @escaping (Action) -> Void
) -> UIAlertController

public static func actionSheet<Action>(
  store: Store<ActionSheetState<Action>, Action>,
  dismiss: Action
) -> UIAlertController

public static func alert<Action>(
  state: AlertState<Action>,
  send: @escaping (Action) -> Void
) -> UIAlertController

public static func alert<Action>(
  store: Store<AlertState<Action>, Action>,
  dismiss: Action
) -> UIAlertController

}
```

Looking forward to any comments.

PS. The same I would suggest to ReactiveSwift Composable.

**Comment by andreyz** [comment]
👋 your suggestion pushed me to finally make a PR with the UIKit helpers I've been using in production here #563. See if that fits your requirements.

**Reply by larryonoff** [reply]
Looks what I need :)
